src/UI/Actions.hs

```diff
@@ -507,10 +508,13 @@ invokeEditor' s = do
   let editor = view (asConfig . confEditor) s
   tmpdir <- getTemporaryDirectory
   tmpfile <- emptyTempFile tmpdir "purebred.tmp"
-  status <- system (editor <> " " <> tmpfile)
+  status <- onException (system (editor <> " " <> tmpfile)) (pure $ setError editorError)
   case status of
-    ExitFailure _ -> pure $ set asAppMode BrowseMail s -- ^ show error XXX
-    ExitSuccess -> pure $ set asAppMode ComposeEditor s & asCompose . cTmpFile ?~ tmpfile -- ^ go to compose editor
+    ExitFailure _ -> pure $ s & set asAppMode BrowseMail & setError editorError
+    ExitSuccess -> pure $ set asAppMode ComposeEditor s & asCompose . cTmpFile ?~ tmpfile -- go to compose editor
```

Comment:
If we're editing this line could we at the same time move from
```haskell
asCompose . cTmpFile ?~ tmpfile 
```
to
```haskell
set (asCompose . cTmpFile) (Just tmpfile)
```
The symbolic lens functions are hard to read.
I can never remember what they mean; what hope do newbies have?

Reply:
Yes. I'm all in.